File: core/src/com/maattss/intro/exercise/states/Task1State.java
package com.maattss.intro.exercise.states;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.math.Rectangle;
import com.maattss.intro.exercise.IntroExercise;
import com.maattss.intro.exercise.sprites.BackButton;
import com.maattss.intro.exercise.sprites.Helicopter;

public class Task1State extends State {
    private Helicopter heli;
    private BackButton backBtn;

    public Task1State(GameStateManager gsm) {
        super(gsm);
        heli = new Helicopter(0, 0);
        backBtn = new BackButton();
    }

    @Override
    protected void handleInput() {
        if(Gdx.input.justTouched()){
            Rectangle touch = new Rectangle(Gdx.input.getX(), IntroExercise.HEIGHT - Gdx.input.getY(),1,1);
            if(touch.overlaps(backBtn.getBounds())) { // User pushed back button
                gsm.set(new MenuState(gsm));
            }
        }
    }

    @Override
    public void update(float dt) {
        handleInput();
        heli.update(dt);
    }

    @Override
    public void render(SpriteBatch sb) {
        sb.begin();
        Gdx.gl.glClearColor(1, 0, 0.996f, 1);
        sb.draw(backBtn.getTexture(), backBtn.getX(), backBtn.getY());
        sb.draw(heli.getHelicopterSprite(),heli.getPosition().x, heli.getPosition().y);
        sb.end();
    }

    @Override
    public void dispose() {
        heli.dispose();
    }
}

File: core/src/com/maattss/intro/exercise/states/Task1State.kt
package com.maattss.intro.exercise.states

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.math.Rectangle
import com.maattss.intro.exercise.IntroExercise
import com.maattss.intro.exercise.sprites.BackButton
import com.maattss.intro.exercise.sprites.Helicopter

class Task1State(gsm: GameStateManager?) : State(gsm) {
    private val heli: Helicopter = Helicopter(0, 0)
    private val backBtn: BackButton = BackButton(false)
    override fun handleInput() {
        if (Gdx.input.justTouched()) {
            val touch = Rectangle(Gdx.input.x.toFloat(),
                    (IntroExercise.HEIGHT - Gdx.input.y).toFloat(), 1f, 1f)
            if (touch.overlaps(backBtn.bounds)) { // User pushed back button
                gsm.set(MenuState(gsm))
            }
        }
    }

    override fun update(dt: Float) {
        handleInput()
        heli.update(dt)
    }

    override fun render(sb: SpriteBatch) {
        sb.begin()
        Gdx.gl.glClearColor(1f, 0f, 0.996f, 1f)
        sb.draw(backBtn.texture, backBtn.x, backBtn.y)
        sb.draw(heli.helicopterSprite, heli.position.x, heli.position.y)
        sb.end()
    }

    override fun dispose() {
        heli.dispose()
    }
}